lib/jekyll-import/importers/dotclear.rb

```diff
@@ -1,121 +1,101 @@
-
-            content = elts[headers[type_data].index("post_content_xhtml")].to_s
-            content = content.gsub('\"', '"').gsub('\n', "\n").gsub("/public/", "/assets/images/")
-
-            filepath = File.join(Dir.pwd, (draft ? "_drafts" : "_posts"), "#{date_str_formatted}-#{title_param}.html")
-
-            entire_content_file = <<~POST_FILE
-              ---
-              layout: post
-              title: "#{elts[headers[type_data].index("post_title")]}"
-              date: #{elts[headers[type_data].index("post_creadt")]} +0100
-              tags: ABC
-              ---
-
-              #{content}
-            POST_FILE
-
-            posts_and_drafts[elts[headers[type_data].index("post_id")]] = { :path => filepath, :content => entire_content_file }
-          elsif type_data == "media"
-            elts[headers[type_data].index("media_title")]
-            mediafilepath = elts[headers[type_data].index("media_file")]
-
-            src_path = File.join(options[:mediafolder], mediafilepath)
-            dst_path = File.join(Dir.pwd, "assets", "images", mediafilepath.to_s)
-
-            FileUtils.mkdir_p(File.dirname(dst_path))
-            FileUtils.cp(src_path, dst_path)
-          elsif type_data == "meta"
-            keywords[elts[headers[type_data].index("post_id")]] ||= []
-            keywords[elts[headers[type_data].index("post_id")]] << elts[headers[type_data].index("meta_id")]
-          elsif type_data == "link"
-
-          elsif type_data == "setting"
-
-          elsif type_data == "comment"
+          file_path = File.expand_path(file_path)
+          if File.open(file_path, "rb", &:readline).start_with?("///DOTCLEAR|")
+            @data = read_export(file_path)
+            Jekyll.logger.info "Export File:", file_path
+          else
+            Jekyll.logger.abort_with "Import Error:", "#{file_path.inspect} is not a valid Dotclear export file!"
+          end
+        end
+        # rubocop:enable Style/IfUnlessModifier
+
+        def process(_opts)
+          Jekyll.logger.info "Importing.."
+          tags = register_post_tags
+          posts = @data["post"]
+          FileUtils.mkdir_p("_drafts") unless posts.empty?
+
+          posts.each do |post|
+            date, title, content = post.values_at("post_creadt", "post_title", "post_content")
+            path = File.join("_drafts", Date.parse(date).strftime("%Y-%m-%d-") + Jekyll::Utils.slugify(title) + ".md")
+            front_matter_data = {
+              "layout" => "post",
+              "title"  => title,
+              "date"   => date,
+              "lang"   => post["post_lang"],
+              "tags"   => tags[post["post_id"]],
+            }.tap(&:compact!)
+
+            # keep a record of existing URL for the post. Jekyll may or may not generate the same
+            # URL depending on `permalink` settings in Jekyll configuration.
+            front_matter_data["dotclear_post_url"] = post["post_url"]
```

Comment:
Could this be unlined into the creation of front_matter_data instead of getting its own declaration? Or are we worried it'll be null?

Reply:
No, we are not worried about this being null since we are calling `compact!` on the hash. This is given a separate declaration to have the comment above it, as to shed light on why we are not setting this as value to a `permalink` front matter key right away.